lib/watir/elements/select.rb

```diff
@@ -54,9 +62,16 @@ def select_all(*str_or_rx)
     # @raise [Watir::Exception::NoValueFoundException] if the value does not exist.
     #
 
-    def select!(*str_or_rx)
-      results = str_or_rx.flatten.map { |v| select_by!(v, :single) }
-      results.first
+    def select!(*both, text: nil, value: nil, label: nil)
+      selection = {both: both, text: text, value: value, label: label}.select do |_k, v|
+        !v.nil? && ![v].flatten.empty?
+      end
+      raise 'Can not select by more than one method' if selection.size > 1
+
+      value = selection.values.first
+      type_check(value)
+
+      [value].flatten.map { |v| select_by! v, :multiple }.first
```

Comment:
Same as for `#select`, we don't seem to actually do anything with the `selection.keys.first`.